coderd/userauth.go

```diff
@@ -1468,8 +1472,36 @@ func (api *API) oauthLogin(r *http.Request, params *oauthLoginParams) ([]*http.C
 				}
 			}
 
+			//nolint:gocritic // No user present in the context.
+			defaultOrganization, err := tx.GetDefaultOrganization(dbauthz.AsSystemRestricted(ctx))
+			if err != nil {
+				// If there is no default org, then we can't assign groups.
+				// By default, we assume all groups belong to the default org.
+				return xerrors.Errorf("get default organization: %w", err)
+			}
+
+			//nolint:gocritic // No user present in the context.
+			memberships, err := tx.GetOrganizationMembershipsByUserID(dbauthz.AsSystemRestricted(ctx), user.ID)
+			if err != nil {
+				return xerrors.Errorf("get organization memberships: %w", err)
+			}
+
+			inDefault := false
+			for _, membership := range memberships {
+				if membership.OrganizationID == defaultOrganization.ID {
+					inDefault = true
+					break
+				}
+			}
+
+			if !inDefault {
+				return xerrors.Errorf("user %s is not a member of the default organization, cannot assign to groups in the org", user.ID)
+			}
+
 			//nolint:gocritic
-			err := api.Options.SetUserGroups(dbauthz.AsSystemRestricted(ctx), logger, tx, user.ID, filtered, params.CreateMissingGroups)
+			err = api.Options.SetUserGroups(dbauthz.AsSystemRestricted(ctx), logger, tx, user.ID, map[uuid.UUID][]string{
+				defaultOrganization.ID: filtered,
+			}, params.CreateMissingGroups)
```

Comment:
I do not love how much code this ended up being. I can reduce it by adding another query, but this is all temporary. We need to solve group mapping to orgs, this code is static to only work with the default org.

This PR is required to remove the single org assumption.